src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -145,6 +209,12 @@ public override void VisitResourceDeclarationSyntax(ResourceDeclarationSyntax sy
                     diagnostics.Write(DiagnosticBuilder.ForPosition(syntax.Type).ResourceTypesUnavailable(resourceType.TypeReference));
                 }
 
+                if (syntax.Value is ForSyntax)
+                {
+                    // TODO: Narrowing doesn't work

```

Comment:
Mind adding a bit more detail to this comment?

Reply:
I will remove it. This was actually fixed.